Jest: Fail tests on failing prop-types

gu-stav opened the pull request:
### What does it do?

> **Note**
> This PR is meant to be a discussion starter.

The PR extends the jest setup so that every thrown prop-type error would result in a 💥  test failure 💥 .

### Why is it needed?

@simotae14 and I are currently cleaning up the frontend tests, which contain many prop-type errors. Once these are cleaned, I'd like to ensure we do not regress on this.

What does everybody think about that?

### How to test it?

Jest will test itself.

joshuaellis commented:
yes. this. 💯  

There are also libraries that do this, they may cover more than we're anticipating with our Regex. It's also a great initiative as I think when we migrate to TS would should continue to throw on typechecking.

gu-stav commented:
@joshuaellis I've checked all 3 of them: one contains a critical bug (I've incorporated the open and unmerged PR in their repository in our PR), and no one uses the other ones, and they are doing the same regex check.

The one that has downloads does, what I've added inline, but they require the plugin to be set up in the package.js file, which I'd like to avoid.

simotae14 commented:
great solution @gu-stav , it is a great first step toward a possible future TS refactoring of the code

remidej reviewed (dismissed):
Sounds good to me :)

ronronscelestes reviewed (approved):
sounds good to me 💃

markkaylor reviewed (dismissed):
🎈

gu-stav commented:
Tests are passing now, so this is ready to merge 🚀

simotae14 reviewed (approved):
thanks @gu-stav for this improvement
